### PR TITLE
Optionally use bazelisk to control bazel version

### PR DESCRIPTION
**Pull request by Katharine**
Bazel has a habit of updating and promptly breaking everything for us (especially on machines where bazel is maintained by the package manager).

Give ourselves more time to track the rapidly changing world of bazel by using [bazelisk](https://github.com/bazelbuild/bazelisk) to manage bazel versions for us.

`bazelisk` is a drop-in replacement for `bazel`, except that it manages bazel versions for you. Specifically, it will use the latest `bazel` unless otherwise specified by either an environment variable or a `.bazelversion` file. It's currently most easily installed using `go get github.com/bazelbuild/bazelisk` but [binary downloads also exist](https://github.com/bazelbuild/bazelisk/releases).

This PR adds a `.bazelversion` file pointing to 0.25.2 (the latest currently-working version) and changes all the scripts in `hack/` and `experiment/` to prefer bazelisk over bazel if available when reinvoking themselves.

/cc @fejta @BenTheElder

### CODE REVIEW

**Comment by Katharine** [comment]
/retest

(network flakes are great)

File: hack/update-bazel.sh [hunk 1]
```diff
@@ -19,22 +19,22 @@ set -o pipefail
...
-    bazel run @io_k8s_test_infra//hack:update-bazel
-  )
-  exit 0
 else
-  (
-    set -o xtrace
-    bazel run @io_k8s_test_infra//hack:update-bazel
-  )
-  exit 0
+  source $(dirname "${BASH_SOURCE}")/find-bazel.sh
```
Comment (fejta) [inline hack/update-bazel.sh]: I am unsupportive of these scripts sourcing other scripts.

File: .bazelversion [hunk 2]
```diff
@@ -0,0 +1 @@
+0.25.2
```
[the remark this answers is not in the document]
Comment (Katharine) [reply .bazelversion]: Not currently, because I haven't re-fixed the typescript thing yet.

File: hack/update-bazel.sh (hunk 1 above)
Comment (Katharine) [reply hack/update-bazel.sh]: Why?

File: .bazelversion (hunk 2 above)
[the remark this answers is not in the document]
Comment (Katharine) [reply .bazelversion]: Hm, never mind, I guess you fixed it somehow?

[the remark this answers is not in the document]
Comment (Katharine) [reply .bazelversion]: Ah, #12877.

File: hack/update-bazel.sh (hunk 1 above)
Comment (fejta) [reply hack/update-bazel.sh]: It is brittle and unnecessary -- soon someone will add something in find-bazel.sh that causes one of these scripts to break in a strange way.

It also encourages increasingly complicated scripts.

If you want to import something, use python or go

Comment (Katharine) [reply hack/update-bazel.sh]: fine, whatever.

**Pull request closed without merging**